pkg/lightning/backend/local/local.go

```diff
@@ -139,11 +131,19 @@ type importMutexState uint32
 
 const (
 	importMutexStateImport importMutexState = 1 << iota
-	importMutexStateFlush
 	importMutexStateClose
-	importMutexStateLocalIngest
+	// importMutexStateReadLock is a special state because in this state we lock engine with read lock
+	// and add isImportingAtomic with this value. In other state, we directly store with the state value.
+	// so this must always the last value of this enum.
+	importMutexStateReadLock
```

Comment:
don't know if we should implement a lock with these many states

such as `type lock struct {mutex, state}`, use mutex to protect the reading/changing of state. and if the current state and requesting state is not tolerable, block the invoker and add it to the waiting queue.

(reinventing the wheels 😵 )

Reply:
Actually, there are only two locked states, write lock for close/state, and read lock for flush.  This enum is merely used for debugging convenience. The `isImporting` atomic is using to simulate `tryLock` mechanism to avoid blocking? 🤔